hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -314,6 +314,12 @@ private boolean writeRecord(HoodieRecord<T> newRecord, Option<HoodieRecord> comb
         recordsWritten++;
       } else {
         recordsDeleted++;
+        // Clear the new location which is set for all records in init()
+        if (needsUpdateLocation()) {
+          newRecord.unseal();
+          newRecord.clearNewLocation();
```

Comment:
The check `needsUpdateLocation()` may be not necessary because Flink also needs it.

Reply:
by right we shouldn't let write handle know anything about index. can you clarify what you meant by "Flink also needs it"?

Reply:
Flink can also impl RLI with deletes, there is no need to add this check.

Reply:
What Danny means is that Flink handles set needUpdateLocation() = false since in flink case the records are pre-tagged. 

I have removed this check.